test/Vegetation/plant_hydraulics_test.jl

```diff
@@ -79,79 +73,64 @@ FT = Float64
 
     # Set system to hydrostatic equilibrium state by setting fluxes to zero, and setting LHS of both ODEs to 0
     function initial_rhs!(F, Y)
+        # Emma: update to do loop
         T0 = FT(0)
-        flux_in_stem = sum(
-            flux.(
-                root_depths,
-                z_stem_midpoint,
-                p_soil0,
-                Y[1],
-                vg_α,
-                vg_n,
-                vg_m,
-                ν,
-                S_s,
-                K_sat_root,
-                K_sat_stem,
-            ),
-        )
-        flux_out_stem = flux(
-            z_stem_midpoint,
-            z_leaf_midpoint,
-            Y[1],
-            Y[2],
-            vg_α,
-            vg_n,
-            vg_m,
-            ν,
-            S_s,
-            K_sat_stem,
-            K_sat_leaf,
-        )
-
-        F[1] = flux_in_stem - T0
-        F[2] = flux_out_stem - T0
+        for i in 1:(n_leaf + n_stem)
+            if i == 1
+                flux_in = sum(
+                    flux.(
+                        z_root_depths,
+                        plant_hydraulics_domain.compartment_midpoints[i],
+                        p_soil0,
+                        Y[i],
+                        vg_α,
+                        vg_n,
+                        vg_m,
+                        ν,
+                        S_s,
+                        K_sat[:root],
+                        K_sat[plant_hydraulics_domain.compartment_labels[i]],
+                    ),
+                )
+            else
+                flux_in = flux(
+                    plant_hydraulics_domain.compartment_midpoints[i - 1],
+                    plant_hydraulics_domain.compartment_midpoints[i],
+                    Y[i - 1],
+                    Y[i],
+                    vg_α,
+                    vg_n,
+                    vg_m,
+                    ν,
+                    S_s,
+                    K_sat[plant_hydraulics_domain.compartment_labels[i - 1]],
+                    K_sat[plant_hydraulics_domain.compartment_labels[i]],
+                )
+            end
+            F[i] = flux_in - T0
+        end
     end
+    # Update initial guess for N compartments
+    soln = nlsolve(initial_rhs!, Vector(-0.03:0.01:0.07))
+    # Update for n comparements (copy code over)
 
-    soln = nlsolve(initial_rhs!, [-0.03, -0.02])
-    p_stem_0 = soln.zero[1]
-    p_leaf_0 = soln.zero[2]
+    p_0 = Vector{Float64}(undef, 11)
+    for i in 1:(n_stem + n_leaf)
+        p_0 = soln.zero[i]
+    end
 
-    S_l_stem_0 =
-        inverse_water_retention_curve(vg_α, vg_n, vg_m, p_stem_0, ν, S_s)
-    S_l_leaf_0 =
-        inverse_water_retention_curve(vg_α, vg_n, vg_m, p_leaf_0, ν, S_s)
+    S_l = inverse_water_retention_curve.(vg_α, vg_n, vg_m, p_0, ν, S_s)
```

Comment:
why not use soln.zero directly here, and remove the code (lines 117-120) where you create and fill p0?